go/multipooler/executor/result_test.go

```diff
@@ -0,0 +1,603 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package executor
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/multigres/multigres/go/pb/query"
+)
+
+// makeRow creates a query.Row from a slice of values (nil values become NULL).
+func makeRow(values ...any) *query.Row {
+	row := &query.Row{Values: make([][]byte, len(values))}
+	for i, v := range values {
+		if v == nil {
+			row.Values[i] = nil
+		} else if b, ok := v.([]byte); ok {
+			row.Values[i] = b
+		} else {
+			row.Values[i] = []byte(v.(string))
+		}
+	}
+	return row
+}
+
+// makeResult creates a query.QueryResult with the given rows.
+func makeResult(rows ...*query.Row) *query.QueryResult {
+	return &query.QueryResult{Rows: rows}
+}
+
+func TestScanRow(t *testing.T) {
+	t.Run("nil row returns error", func(t *testing.T) {
+		var s string
+		err := ScanRow(nil, &s)
+		require.Error(t, err)
+		assert.Contains(t, err.Error(), "row is nil")
+	})
+
+	t.Run("not enough columns returns error", func(t *testing.T) {
+		row := makeRow("value1")
+		var s1, s2 string
+		err := ScanRow(row, &s1, &s2)
+		require.Error(t, err)
+		assert.Contains(t, err.Error(), "not enough columns")
+	})
+
+	t.Run("scan multiple values successfully", func(t *testing.T) {
+		row := makeRow("hello", "42", "t")
+		var s string
+		var i int
+		var b bool
+		err := ScanRow(row, &s, &i, &b)
+		require.NoError(t, err)
+		assert.Equal(t, "hello", s)
+		assert.Equal(t, 42, i)
+		assert.True(t, b)
+	})
+
+	t.Run("extra columns are ignored", func(t *testing.T) {
+		row := makeRow("hello", "world", "extra")
+		var s1, s2 string
+		err := ScanRow(row, &s1, &s2)
+		require.NoError(t, err)
+		assert.Equal(t, "hello", s1)
+		assert.Equal(t, "world", s2)
+	})
+}
+
+func TestScanSingleRow(t *testing.T) {
+	t.Run("nil result returns error", func(t *testing.T) {
+		var s string
+		err := ScanSingleRow(nil, &s)
+		require.Error(t, err)
+		assert.Contains(t, err.Error(), "no rows")
+	})
+
+	t.Run("empty result returns error", func(t *testing.T) {
+		result := &query.QueryResult{Rows: []*query.Row{}}
+		var s string
+		err := ScanSingleRow(result, &s)
+		require.Error(t, err)
+		assert.Contains(t, err.Error(), "no rows")
+	})
+
+	t.Run("scans first row successfully", func(t *testing.T) {
+		result := makeResult(makeRow("first"), makeRow("second"))
+		var s string
+		err := ScanSingleRow(result, &s)
+		require.NoError(t, err)
+		assert.Equal(t, "first", s)
+	})
+}
+
+func TestScanValue_Bool(t *testing.T) {
+	tests := []struct {
+		name        string
+		input       string
+		expected    bool
+		expectError bool
+	}{
+		{"t", "t", true, false},
```

Comment:
For these error test cases, might be nice to have the expected error message instead of just a boolean for whether or not an error happened. Same comment for other similar test cases in this file

Reply:
Fixed